json-everything.net/wwwroot/md/vocabs-data-2022.md

```diff
@@ -0,0 +1,123 @@
+# A Vocabulary for Accessing Data Stored in JSON
+
+## 1. Purpose
+
+This document describes a vocabulary defining keywords that can be used to reference values stored in
+
+- the instance data
+- external JSON data
+- the schema data
+
+and use the dereferenced values as input for other keywords.
+
+The intent for this keyword is to cover the use cases discussed across several issues in the JSON Schema specification GitHub repositories.  (A quick search for `"$data"` can readily summon these issues.)
+
+## 2. Declarations
+
+The ID for this vocabulary is `https://json-everything.net/vocabs-data-2022` (the URI to this document).
+
+A draft 2020-12 meta-schema which includes this vocabulary has been defined for convenience.  The ID for the meta-schema is `https://json-everything.net/meta/data-2022`, and it can also be found at this address.
+
+## 3. The `data` Keyword
+
+### 3.1 Syntax and Semantics
+
+The value of `data` must be an object.  The keys of the object correspond to valid JSON Schema keywords, and the values MUST be one of
+
+- JSON Pointers per [RFC 6901](https://www.rfc-editor.org/rfc/rfc6901)
+- [Relative JSON Pointers](https://json-schema.org/draft/2019-09/relative-json-pointer.html)
+- URI references per [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986), potentially with JSON Pointer fragments
+
+The set of keywords in the schema or subschema containing `data` SHOULD be distinct from the set of keys defined within `data`.  Behavior in the event of an overlap is not defined.
```

Comment:
The original intent was that rather than `data` operating as an applicator (as we've discussed), the result would be "merged" with the parent schema.  That would mean that keyword conflicts could arise.

I think `data` operating as an applicator resolves this, so, yeah, this could just come out.

Reply:
Awesome - yeah it's interesting that neither of us looked at `data` as an applicator at first (I only called it "kind of like an applicator"), but once we agreed on that way of looking at it things got simpler all around.  That's really the kind of thing I've always hoped the classifications would help with.